include/exadg/incompressible_navier_stokes/spatial_discretization/spatial_operator_base.cpp

```diff
@@ -317,24 +340,9 @@ SpatialOperatorBase<dim, Number>::distribute_dofs()
     }
   }
 
-  unsigned int ndofs_per_cell_velocity;
-  if(param.spatial_discretization == SpatialDiscretization::L2)
-  {
-    ndofs_per_cell_velocity = dealii::Utilities::pow(param.degree_u + 1, dim) * dim;
-  }
-  else if(param.spatial_discretization == SpatialDiscretization::HDIV)
-  {
-    ndofs_per_cell_velocity =
-      dealii::Utilities::pow(param.degree_u, dim - 1) * (param.degree_u + 1) * dim;
-  }
-  else
-  {
-    AssertThrow(false, dealii::ExcMessage("FE not implemented."));
-  }
-
-  unsigned int const ndofs_per_cell_pressure =
-    dealii::Utilities::pow(param.get_degree_p(param.degree_u) + 1, dim);
+  unsigned int ndofs_per_cell_velocity = fe_u->n_dofs_per_cell();
```

Comment:
This might not work for HDIV, see: https://github.com/exadg/exadg/pull/349#discussion_r1129664955

Reply:
This should still work. The other comment was about using something like `fe_u->base_element(0)` to return the element of a scalar component, as all components are within a single monolithic system in Hdiv. (There is a separation in `FE_RaviartThomasNodal` from the polynomial point of view, but the final polynomials cover the whole space and the Piola transform couples them together.)